cmd/ocm/create/idp/htpasswd.go

```diff
@@ -0,0 +1,64 @@
+/*
+Copyright (c) 2020 Red Hat, Inc.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+  http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package idp
+
+import (
+	"errors"
+	"fmt"
+
+	cmv1 "github.com/openshift-online/ocm-sdk-go/clustersmgmt/v1"
+
+	"github.com/AlecAivazis/survey/v2"
+)
+
+func buildHtpasswdIdp(cluster *cmv1.Cluster, idpName string) (idpBuilder cmv1.IdentityProviderBuilder, err error) {
+	username := args.htpasswdUsername
+	password := args.htpasswdPassword
+
+	if username == "" {
+		prompt := &survey.Input{
+			Message: "Enter username:",
+		}
+		err = survey.AskOne(prompt, &username)
+		if err != nil {
+			return idpBuilder, errors.New("Expected a username")
+		}
+	}
+
+	if password == "" {
+		prompt := &survey.Password{
+			Message: "Enter password:",
+		}
+		err = survey.AskOne(prompt, &password)
+		if err != nil {
+			return idpBuilder, errors.New("Expected a password")
+		}
+	}
+
+	// Create HTPasswd IDP
+	htpasswdIDP := cmv1.NewHTPasswdIdentityProvider().
+		Username(username).
+		Password(password)
+
+	// Create new IDP with HTPasswd provider
+	idpBuilder.
+		Type("HTPasswdIdentityProvider"). // FIXME: ocm-api-model has the wrong enum values
+		Name(idpName).
+		MappingMethod(cmv1.IdentityProviderMappingMethod(args.mappingMethod)).
+		Htpasswd(htpasswdIDP)
+
+	fmt.Println("Securely store your username and password.")
```

Comment:
Thanks, @NautiluX . Overall LGTM. I would add this after IDP is successfully created so it doesn't get lost in the trace 

https://github.com/openshift-online/ocm-cli/blob/85ebe2ebca0465b978e98eb36a2ae3f917e6b78c/cmd/ocm/create/idp/cmd.go#L347

Can we expand IDP to either identity provider or this is what we have currently in ROSA 
`
Please securely store this generated password. If you lose this password you can delete and recreate the cluster admin user.`

Reply:
Do you suggest to only allow generated passwords rather than getting one as input? The UI allows to specify a password, so I thought it would be okay to allow it as input. We could use a generated one if none is entered?
For the output, the current implementation doesn't allow IDP-specific output after the creation, so I added it before, I think doing that without a `if htpasswd` would involve quite some refactoring. Do you think the order of log lines is worth the refactoring?

Reply:
@NautiluX  no :)  I just wanted to move the log lines

fmt.Println("Securely store your username and password.")
fmt.Println("If you lose these credentials, you will have to delete and recreate the IDP.")

 after we successfully created the idp in  cmd/ocm/create/idp/cmd.go#L353

Reply:
uhm ok, now i added the generation option... You like it? :D
I'm not sure I like how htpasswd is handled differently now regarding the additional log lines.